### PR TITLE
borrowck: name the correct type in error message 

### LINKED ISSUE #36407: Confusing error message wording for moving out of non-copy slices.
https://is.gd/iH1SwT

``` rust
let x = vec![String::from("foo")].into_boxed_slice();
let y = x[0];

// error: cannot move out of type `[std::string::String]`, a non-copy fixed-size array [--explain E0508]
```

It refers to the slice as a fixed-sized array. This can also be reproduced with a `Vec<String>` by [indexing `(*x)[0]`](https://is.gd/rK0ddS) instead of [indexing `x[0]`](https://is.gd/5RW9Qy).

### PR DESCRIPTION
**Pull request by tamird**
Closes #36407.

r? @Mark-Simulacrum

### CODE REVIEW

**Comment by arielb1** [comment]
But this error also occurs for arrays:
```Rust
fn main() {
    let x = [Box::new(0)][0];
}
```

```
error[E0508]: cannot move out of type `[std::boxed::Box<i32>; 1]`, a non-copy array
 --> src/main.rs:2:13
  |
2 |     let x = [Box::new(0)][0];
  |             ^^^^^^^^^^^^^^^^
  |             |
  |             help: consider using a reference instead `&[Box::new(0)][0]`
  |             cannot move out of here
```

**Comment by arielb1** [comment]
The type is available right here on `b.ty.sty`.

**Comment by tamird** [comment]
This should be good to go.

File: src/librustc_borrowck/borrowck/gather_loans/move_error.rs [hunk 1]
```diff
@@ -153,20 +153,26 @@ fn report_cannot_move_out_of<'a, 'tcx>(bccx: &BorrowckCtxt<'a, 'tcx>,
...
-                                                    a non-copy array",
-                                                   b.ty);
-                    err.span_label(move_from.span, "cannot move out of here");
-                    err
-                }
-                (_, Kind::Pattern) => {
-                    span_bug!(move_from.span, "this path should not cause illegal move");
-                }
-            }
+            let type_name = match &b.ty.sty {
```
Comment (arielb1) [inline src/librustc_borrowck/borrowck/gather_loans/move_error.rs]: Can you maintain the `bug!` on `b.ty.sty == ty::TyArray` and `ik == Kind::Pattern`?

Comment (tamird) [inline src/librustc_borrowck/borrowck/gather_loans/move_error.rs]: Sure, done.

**Pull request merged**